nni/retiarii/strategy/evolution.py

```diff
@@ -0,0 +1,157 @@
+import collections
+import dataclasses
+import logging
+import random
+import time
+
+from ..execution import query_available_resources, submit_models
+from ..graph import ModelStatus
+from .base import BaseStrategy
+from .utils import dry_run_for_search_space, get_targeted_model
+
+
+_logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Individual:
+    """
+    A class that represents an individual.
+    Holds two attributes, where ``x`` is the model and ``y`` is the metric (e.g., accuracy).
+    """
+    x: dict
+    y: float
+
+
+class RegularizedEvolution(BaseStrategy):
+    """
+    Algorithm for regularized evolution (i.e. aging evolution).
+    Follows "Algorithm 1" in Real et al. "Regularized Evolution for Image Classifier Architecture Search".
+
+    Parameters
+    ----------
+    optimize_mode : str
+        Can be one of "maximize" and "minimize".
+    population_size : int
+        The number of individuals to keep in the population.
+    cycles : int
+        The number of cycles (trials) the algorithm should run for.
```

Comment:
cycle is a little misleading, cycle means trial in the paper?

Reply:
The terminology is used in paper and their open-source implementation.

![image](https://user-images.githubusercontent.com/8463288/108690378-40741d00-7535-11eb-958e-953698821667.png)

Reply:
ok